code-actions-sample/src/diagnostics.ts

```diff
@@ -23,21 +23,25 @@ export function refreshDiagnostics(doc: vscode.TextDocument, emojiDiagnostics: v
 
 	for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
 		const lineOfText = doc.lineAt(lineIndex);
-		if (lineOfText.text.includes(EMOJI)) {
-			diagnostics.push(createDiagnostic(doc, lineOfText, lineIndex));
+		const regex = RegExp(EMOJI, 'g');
+		let arr;
+		while ((arr = regex.exec(lineOfText.text)) !== null) {
```

Comment:
If you also update minimum VS Code engine from 1.32 to at least 1.40 you can use `matchAll`, which I think looks cleaner.

```suggestion
		const matches = lineOfText.text.matchAll(regex);
		for (const arr of matches) {
```

Reply:
thanks. I do it

Reply:
Indentation looks off for the inner `for`.

Reply:
done, it look that because I use spaces for indent